clippy_lints/src/loops/needless_range_loop.rs

```diff
@@ -107,7 +107,7 @@ pub(super) fn check<'tcx>(
                                 take_expr = left;
                             }
 
-                            end_is_start_plus_val = start_equal_left | start_equal_right;
+                            end_is_start_plus_val = start_equal_left || start_equal_right;
```

Comment:
This is a prime example I personally wouldn't want to lint. It's very likely not improving perf to short-circuit here.